### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by ericelliott**
@Mouvedia comments? Specifically, thoughts on `class`?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -210,24 +210,18 @@ The special type `Any` means that any type is allowed:
 
 #### The `Void` Type
 
-The special type `Void` should only be used to indicate that a function returns no meaningful value (i.e., `undefined`). Since `Void` is the default return type, it can be optionally omitted. Nevertheless `Void` return types *should* usually be explicitly annotated to denote function side-effects.
+The special type `Void` should only be used to indicate that a function returns no meaningful value (i.e., implicit `undefined`).
```
Comment (Mouvedia) [inline README.md:213]: Is it an undocumented feature from now on?
The only merit is that it would open up the possibility of having a different default return type for constructors/classes.

cf #110

File: README.md [hunk 2]
```diff
@@ -128,7 +128,7 @@ Optionally, you may name the return value, similar to named parameters:
 Or even name a signature to reuse it later on:
 
 ```js
-connect(options: Object) => connection: Object
+connect = (options: Object) => connection: Object
```
Comment (Mouvedia) [inline README.md:131]: If we had tools depending on this README that would be a breaking change; I think those changes should be proposed on a separate PR.

File: README.md [hunk 3]
```diff
@@ -261,29 +255,30 @@ It covers these contructors: `Int8Array`, `Uint8Array`, `Uint8ClampedArray`, `In
...
-The type of arrays' elements can also be specified:
+The type of arrays elements can also be specified:
 
 ```js
 // an array that contains exactly 2 elements
 [Number, String]
 ```
 
-For **∅ or more** and **1 or more** element(s) of the same type you can use the rest operator like so:
+For **0 or more** and **1 or more** element(s) of the same type you can use the rest operator like so:
```
Comment (Mouvedia) [inline README.md:273]: What about array holes? I think ∅ covers more cases.

File: README.md [hunk 4]
```diff
@@ -293,18 +288,18 @@ For **∅ or more** and **1 or more** element(s) of the same type you can use th
...
 
 ```js
-(userInput: String|Number) => String|Number
+(userInput: String | Number) => String | Number
 ```
 
 ### Constructors
 
-Constructors in JavaScript require the `new` keyword. You can identify a constructor signature using the `new` keyword as if you were demonstrating usage:
+Constructors and classes are denoted using the `class` keyword:
```
Comment (Mouvedia) [inline README.md:296]: If you know JS, you can tell that classes are just a convenience glorified syntax.
As you can tell, I am not a big fan of `class` so I am a bit biased on this: Ill let you decide.
Is it an effort to be more compatible with TS?

If you wanna move on with that change the title will have to be changed.
You gotta make a clear distinction between the constructor statement of a class and what you call a "constructor".

You ought to provide at least 2 examples:
- constructor function
- class

File: README.md [hunk 5]
```diff
@@ -318,10 +313,10 @@ interface UserInstance {
 An accessor function is defined by prefixing a method with `get` or `set`.
 
 ```js
-new User({ username: String }) => {
+class User({ username: String }) => {
   username: String,
   get name() => String,
-  set name(newName: String) // return type defaults to Void
+  set name(newName: String) => Void
```
Comment (Mouvedia) [inline README.md:319]: I am opposed to this change.

**Review by Mouvedia — CHANGES_REQUESTED** [review]
new PR for `=`
`class` your call
nay on the setter change

File: README.md (hunk 3 above)
Comment (ericelliott) [reply README.md:273]: OK

File: README.md (hunk 5 above)
Comment (ericelliott) [reply README.md:319]: OK

File: README.md (hunk 4 above)
Comment (ericelliott) [reply README.md:296]: I think this deserves more thought. Thanks for your feedback on all this. =)
